### PR TITLE
Fix/interfaces

### PR DESCRIPTION
**Pull request by Manuthor**

### CODE REVIEW

File: src/interfaces/ffi/tests.rs [hunk 1]
```diff
@@ -239,9 +240,9 @@ unsafe fn encrypt_header_using_cache(
     header_metadata: &[u8],
     authentication_data: &[u8],
 ) -> Result<(<DEM as Dem<{ DEM::KEY_LENGTH }>>::Key, EncryptedHeader), Error> {
-    let policy_cs = CString::new(serde_json::to_string(&policy)?.as_str())
-        .map_err(|e| Error::Other(e.to_string()))?;
-    let policy_ptr = policy_cs.as_ptr();
+    let policy_bytes = serde_json::to_vec(&policy)?;
```
Comment (ThibaudDauce) [inline src/interfaces/ffi/tests.rs:243]: `serde_json::to_vec` is just JSON then UTF8 bytes? I prefer to have it explicitly in two steps than hidding it in `serve_json::to_vec`

Comment (tbrezot) [reply src/interfaces/ffi/tests.rs:243]: will be done in next release

File: src/interfaces/ffi/hybrid_cc_aes.rs [hunk 2]
```diff
@@ -974,90 +645,20 @@ pub unsafe extern "C" fn h_aes_decrypt(
...
-        set_last_error(FfiError::Generic(format!(
-            "access policy to JSON: the pre-allocated buffer is too small. It should be {} bytes \
-             long",
-            bytes.len()
-        )));
-        return bytes.len() as i32;
+    if metadata_ptr.is_null() || *metadata_len == 0 {
+        let null_ptr = null_mut::<c_char>();
+        let null_len = &mut 0;
+        ffi_write_bytes!("metadata", &decrypted_header.metadata, null_ptr, null_len);
```
Comment (ThibaudDauce) [inline src/interfaces/ffi/hybrid_cc_aes.rs:653]: Not sure to understand this part? Maybe add a comment?

Comment (tbrezot) [reply src/interfaces/ffi/hybrid_cc_aes.rs:653]: I can add a comment.

**Pull request merged**